metrics/scaling/k8s_scale_net.sh

```diff
@@ -0,0 +1,287 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+# Pull in some common, useful, items
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+source "${SCRIPT_PATH}/common.bash"
+
+LABELVALUE=${LABELVALUE:-scale_net}
+
+# Set some default metrics env vars
+TEST_ARGS="runtime=${RUNTIME}"
+TEST_NAME="k8s scaling net"
+input_yaml="${SCRIPT_PATH}/net-serve.yaml.in"
+input_json="${SCRIPT_PATH}/net-serve.json.in"
+name_base_depl="net-serve"
+
+# $1 is the launch time in seconds this pod/container took to start up.
+# $2 is the number of pod/containers under test
+# $3 is the time to pod network measure
+grab_stats(){
+    local launch_time_ms=$1
+    local n_pods=$2
+    local net_time=$3
+
+    info "And grab some stats"
+
+    local date_json="$(cat << EOF
+        "date": {
+            "ns": $(date +%s%N),
+            "Date": "$(date -u +"%Y-%m-%dT%T.%3N")"
+        }
+EOF
+)"
+    metrics_json_add_array_fragment "$date_json"
+
+    local pods_json="$(cat << EOF
+	                "n_pods": {
+                            "Result": ${n_pods},
+                            "Units" : "int",
+                            "time_to_pod_net": ${net_time},
+                            "Units" : "ms"
+                    }
+EOF
+)"
+    metrics_json_add_array_fragment "$pods_json"
+
+    local launch_json="$(cat << EOF
+                        "launch_time": {
+                                "Result": $launch_time_ms,
+                                "Units" : "ms"
+                            }
+EOF
+)"
+    metrics_json_add_array_fragment "$launch_json"
+
+    info "launch [$launch_time_ms]"
+
+    metrics_json_close_array_element
+}
+
+init() {
+    info "Initialising"
+
+    local cmds=("bc" "jq")
+    check_cmds "${cmds[@]}"
+
+    info "Checking Kubernetes accessible"
+    local worked=$( kubectl get nodes > /dev/null 2>&1 && echo $? || echo $? )
+    if [ "$worked" != 0 ]; then
+        die "kubectl failed to get nodes"
+    fi
+
+    info $(get_num_nodes) "Kubernetes nodes in 'Ready' state found"
+
+    k8s_api_init
```

Comment:
now #268 has merged, I think you'll need to rebase and use the `framework_init` type calls - so we can optionally enable othe r features (like the cpu load). Have a read of the patches in #268 to see what needs updating (where similar tests were patched).

Reply:
The code is rebased and after reviewing the details of `framework_init`, I decided to use `k8s_api_init` to avoid any noise that [`cpu_load_init`](https://github.com/clearlinux/cloud-native-setup/blob/master/metrics/lib/common.bash#L98) may add.

Reply:
nack. The whole idea is that *all* tests call framework init, and then 'features', like the `cpu_load_init` can be *optionally* turned on and off at will.
By default the cpu-load code should be disabled for this test, unless it has been specifically enabled. See https://github.com/clearlinux/cloud-native-setup/blob/master/metrics/lib/cpu-load.bash#L93

Reply:
Done.